docs/adr/adr_003_bundling_collection_data.md

```diff
@@ -0,0 +1,209 @@
+ - `fakespot-suggest-products` collection: 300 records with a few MB per attachment (eg. 5MB). Since the total size is huge: bundling brings __little advantage__
+ - Machine learning collections: few records with big attachments (eg. 20MB): almost no advantage. 
+ - `Translations-models` collection: 150 records for 1.2GB in total. Clients only download a few pairs when the user enables the feature for certain languages: __bundling is not feasible__
+
+That’s why it would make sense for attachments bundling to be opt-in for stakeholders.
+
+There is already a collection metadata attribute to mark attachments as enabled and/or required. The flag would just consist in adding a new field attachment.bundle: true|false.
+
+Also, since [usage quotas](https://docs.kinto-storage.org/en/stable/api/1.x/quotas.html) are not enabled on the Remote Settings server, consumers can end up publishing huge files and bloat attachments bundles, we may want to have a server side configuration setting to fix a size limit for which the bundles won’t be built. For example, even when opted-in, if the target bundle exceeds 20MB, it won’t be built and published. Clients should thus be smart and fail soft if a bundle is not available.
+
+
+### Bundle Formats
+After some research and testing, we've decided to go with Zip compression. Zip is very easy to implement and allows for good flexibility moving forward. We could change our minds or add other bundle types in the future with very little effort.
+
+- Zip - Reduces text content up to 17%.
+    - Pros: Easy to work with. Easy to decompress client side.
+    - Cons: Not as much compression as lzma. Need to decompress after receiving.
+- Tar - When gzipped (automatically by cdn dynamic compression), we can expect to save around 43% size for text content. Probably not much for images.
+    - Pros: very easy to work with server side and client side. Compression and decompression should be free at the transport layer.
+    - Cons: doesn’t save as much as lzma.
+- LZMA (7z) - Reduces text content up to 70%.
+    - Pros: Very compressed
+    - Cons: More compute intense to compress and decompress. Likely need to ship another library to decompress. Need to decompress after receiving.
+
+
+### Client Implementation
+This ADR mostly covers the server parts, since the client implementation to fetch and leverage bundling does not have many different possible approaches.
+
+For the collections records bundle, clients with empty profiles will simply pull it on first synchronization:
+- Pull the latest records bundle
+- Extract it and store data into local IndexedDB
+- Fetch latest timestamps from the server (like currently using `Utils.fetchLatestChanges()`)
+- Synchronize potential outdated collections, and verify signatures of the ones that are up-to-date, to guarantee integrity/authenticity of the downloaded bundle
```

Comment:
Having content signatures on the bundles could have some benefits indeed. 
However, it would require the cronjob to have access to Autograph.

With the proposed approach, the bundle itself is not verified but:
- we verify the signature of the local data for each collection after its extraction
- the signature of the cached attachment is verified when `client.attachments.download()` is called

So, if the bundle is corrupted, the local extracted data and attachments would be erased and refetched from server if signature verification fails.

We could have a `.md5sum` file though, just to verify the basic integrity of the downloaded file before trying to extract it.

Reply:
I suggest that we put md5 sums as a possible future item. They could be helpful for resumable downloads to know if the bundle changed.